code/modules/assembly/flash.dm

```diff
@@ -281,4 +281,8 @@
 		addtimer(CALLBACK(src, .proc/update_icon), 5)
 
 	if(holder)
-		holder.update_icon()
\ No newline at end of file
+		holder.update_icon()
+
+/obj/item/device/assembly/flash/shield/hit_reaction(obj/item/weapon/W, mob/user, params)
+	activate()

```

Comment:
Yes. But I can't see a way to get if the attack was blocked from hit_reaction. This does give them a weakness in that someone with glasses can spam you and burn out your shield.